layers/+distributions/spacemacs-base/local/hybrid-mode/hybrid-mode.el

```diff
@@ -118,12 +118,16 @@
   :exit-hook (evil-cleanup-insert-state evil-stop-track-last-insertion)
   :input-method t
   (cond
-   ((evil-hybrid-state-p)
+   ((evil-insert-state-p)
```

Comment:
I was surprised by this too, but `evil-insert-state-p` is overridden in `hybrid-mode.el` which is why I think it works.